network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py

```diff
@@ -238,6 +241,39 @@ def entries_by_topic(self, request, topic, *args, **kwargs):
 
         return IndexPage.serve(self, request, *args, **kwargs)
 
+    @route(r'^authors/$')
+    def blog_author_index(self, request: HttpRequest, *args, **kwargs) -> 'HttpResponse':
+        author_profiles = Profile.objects.all()
+        author_profiles = author_profiles.filter_blog_authors()
+        author_profiles = localize_queryset(author_profiles)
+
+        return self.render(
+            request,
+            context_overrides={
+                'title': 'Blog authors',
+                'author_profiles': author_profiles
+            },
+            template="wagtailpages/blog_author_index_page.html"
+        )
+
+    @route(r'^authors/(?P<profile_id>.+)/', name='blog-author-detail')
+    def blog_author_detail(self, request: HttpRequest, profile_id: str, *args, **kwargs) -> 'HttpResponse':
+        try:
+            author_profile = get_object_or_404(Profile, id=profile_id)
+        except ValueError:
+            author_profile = None
+        if not author_profile:
+            raise Http404()
```

Comment:
**Question** Isn't this the same as only `author_profile = get_object_or_404(Profile, id=profile_id)`?

Reply:
you're right - I added this before - will fix

Reply:
I've simplified this in 6ab09672c8f9b526af8e4bf8f7731df8632d968a

I still want to catch the ValueError just in case something like `string` makes its way through as a query, because `get_object_or_404` will raise it [ref](https://github.com/django/django/blob/main/django/shortcuts.py#L80) Another option would be to try casting `int(profile_id)` but I prefer catching what `get_object_or_404` is specifically raising